app/enrichment/index.js

```diff
@@ -1,10 +1,12 @@
 const enrichHeader = require('./enrich-header')
+const validatePaymentRequestType = require('./validate-type')
 const validatePaymentRequest = require('./validate-header')
 const processInvoiceLines = require('./process-invoice-lines')
 const validateValues = require('./validate-values')
 const getScheme = require('./get-scheme')
 
 const enrichPaymentRequest = async (paymentRequest) => {
```

Comment:
Is it **guaranteed** that if there isn't a `schemeCode` in the invoice lines that a `standardCode` is?

The func `enrichInvoiceLine` will error out if `await getSchemeCode(invoiceLine.standardCode)` does so on `undefined` (but this is only executed if `invoiceLine.schemeCode` isn't given

Reply:
It is absolutely not guaranteed.  We should assume that our customers may send erroneous data.

I've updated the function to handle a missing standard code and added tests to cover that behaviour.

I've also changed it to cache the standard codes like we do with scheme codes because when we're processing high volumes of requests, we can improve performance by not having to make that DB call for every request.